SocialMediaDB.sql

```diff
@@ -154,9 +154,149 @@ CREATE TABLE CAUSE_COMMUNITY  (
     FOREIGN KEY (page_id) REFERENCES PAGE(page_id) ON DELETE CASCADE ON UPDATE CASCADE
 );
 
-DROP TABLE IF EXISTS SOCIAL_MEDIA.GROUP;
-CREATE TABLE SOCIAL_MEDIA.GROUP (
+DROP TABLE IF EXISTS social_media.GROUP;
+CREATE TABLE social_media.GROUP (
     group_id int AUTO_INCREMENT PRIMARY KEY,
     group_name varchar(50) NOT NULL,
     group_privacy ENUM('Public', 'Private', 'Secret')
-);
\ No newline at end of file
+);
+-- relationships
+
+DROP TABLE IF EXISTS COMMENTS;
+CREATE TABLE COMMENTS (
+    comment_id INT NOT NULL PRIMARY KEY,
+    user_id INT NOT NULL,
+    post_id INT NOT NULL,
+    FOREIGN KEY (comment_id) REFERENCES COMMENT(comment_id) ON DELETE CASCADE ON UPDATE CASCADE,
+    FOREIGN KEY (user_id) REFERENCES USER(user_id) ON DELETE CASCADE ON UPDATE CASCADE,
+    FOREIGN KEY (post_id) REFERENCES POST(post_id) ON DELETE CASCADE ON UPDATE CASCADE
+);
+
+DROP TABLE IF EXISTS FOLLOWS;
+CREATE TABLE FOLLOWS (
+    follower_id INT NOT NULL ,
+    following_id INT NOT NULL,
+    FOREIGN KEY (follower_id) REFERENCES USER(user_id) ON DELETE CASCADE ON UPDATE CASCADE,
+    FOREIGN KEY (following_id) REFERENCES USER(user_id) ON DELETE CASCADE ON UPDATE CASCADE,
+    PRIMARY KEY(follower_id,following_id)
+);
+
+DROP TABLE IF EXISTS MAKES_GENERAL_REACT;
+CREATE TABLE MAKES_GENERAL_REACT (
+    user_id INT NOT NULL ,
+    post_id INT NOT NULL,
+    FOREIGN KEY (user_id) REFERENCES USER(user_id) ON DELETE CASCADE ON UPDATE CASCADE,
+    FOREIGN KEY (post_id) REFERENCES POST(post_id) ON DELETE CASCADE ON UPDATE CASCADE,
+    PRIMARY KEY(user_id,post_id),
+    reacted_type ENUM('Like', 'Haha', 'Heart', 'Angry', 'Wow', 'Dislike')
+);
+
+DROP TABLE IF EXISTS LIKES;
+CREATE TABLE LIKES (
+    user_id INT NOT NULL ,
+    page_id INT NOT NULL,
+    FOREIGN KEY (user_id) REFERENCES USER(user_id) ON DELETE CASCADE ON UPDATE CASCADE,
+    FOREIGN KEY (page_id) REFERENCES PAGE(page_id) ON DELETE CASCADE ON UPDATE CASCADE,
+    PRIMARY KEY(user_id,page_id)
+);
+
+DROP TABLE IF EXISTS BELONGS_TO;
+CREATE TABLE BELONGS_TO (
+    user_id INT NOT NULL ,
+    group_id INT NOT NULL,
+    FOREIGN KEY (user_id) REFERENCES USER(user_id) ON DELETE CASCADE ON UPDATE CASCADE,
+    FOREIGN KEY (group_id) REFERENCES social_media.GROUP(group_id) ON DELETE CASCADE ON UPDATE CASCADE,
+    PRIMARY KEY(user_id,group_id)
+);
+
+DROP TABLE IF EXISTS IS_ADMIN;
+CREATE TABLE IS_ADMIN (
+    user_id INT NOT NULL ,
+    group_id INT NOT NULL,
+    FOREIGN KEY (user_id) REFERENCES USER(user_id) ON DELETE CASCADE ON UPDATE CASCADE,
+    FOREIGN KEY (group_id) REFERENCES social_media.GROUP(group_id) ON DELETE CASCADE ON UPDATE CASCADE,
```

Comment:
Do we actually have to use `social_media.` here?

Reply:
It is not working otherwise. Probably group is a keyword

Reply:
For some weird reason, yes... It isnt working otherwise... Even for creation... Probably cause group signifies something in SQL